compiler/damlc/daml-ide-core/src/Development/IDE/Core/Rules/Daml.hs

```diff
@@ -516,6 +516,25 @@ generatePackageMapRule opts = do
         let hash = BS.concat $ map (T.encodeUtf8 . LF.unPackageId . LF.dalfPackageId) $ Map.elems res
         return (Just hash, ([], Just res))
 
+damlGhcSessionRule :: Options -> Rules ()
+damlGhcSessionRule opts@Options{..} = do
+    defineEarlyCutoff $ \DamlGhcSession projectRoot -> do
+        let base = mkBaseUnits (optUnitId opts)
+        optPackageImports <-
+          if getInferDependantPackages optInferDependantPackages
+          then do
+            more <- liftIO $ dependantUnitsFromDamlYaml optDamlLfVersion (fromNormalizedFilePath projectRoot)
+            pure $ map mkPackageFlag (base ++ more) ++ optPackageImports
+          else
+            pure $ map mkPackageFlag base ++ optPackageImports
```

Comment:
suggestion:

```suggestion
        packages <- if getInferDependentPackages optInferDependantPackages
            then fmap (base ++) . liftIO $ dependentUnitsFromDamlYaml optDamlLfVersion (fromNormalizedFilePath projectRoot)
            else pure base   
        let optPackageImports = map mkPackageFlag packages ++ optPackageImports
```

Reply:
Sorry this is still a draft PR, mostly just testing CI. A bunch of stuff is going to change and needs cleanup :slightly_smiling_face:  I’ll ping you once it’s ready

Reply:
Addressed is as part of cleanup, thanks for the suggestion!